module/purpose/Defender/src/Defender.cpp

```diff
@@ -0,0 +1,103 @@
+#include "Defender.hpp"
+
+#include "extension/Behaviour.hpp"
+#include "extension/Configuration.hpp"
+
+#include "message/input/GameState.hpp"
+#include "message/planning/KickTo.hpp"
+#include "message/purpose/Defender.hpp"
+#include "message/strategy/AlignBallToGoal.hpp"
+#include "message/strategy/FindFeature.hpp"
+#include "message/strategy/KickToGoal.hpp"
+#include "message/strategy/LookAtFeature.hpp"
+#include "message/strategy/Ready.hpp"
+#include "message/strategy/StandStill.hpp"
+#include "message/strategy/WalkInsideBoundedBox.hpp"
+#include "message/strategy/WalkToBall.hpp"
+#include "message/strategy/WalkToFieldPosition.hpp"
+
+#include "utility/support/yaml_expression.hpp"
+
+namespace module::purpose {
+
+    using extension::Configuration;
+    using Phase    = message::input::GameState::Data::Phase;
+    using GameMode = message::input::GameState::Data::Mode;
+    using message::input::GameState;
+    using message::planning::KickTo;
+    using message::purpose::NormalDefender;
+    using message::strategy::AlignBallToGoal;
+    using message::strategy::FindBall;
+    using message::strategy::KickToGoal;
+    using message::strategy::LookAtBall;
+    using message::strategy::Ready;
+    using message::strategy::StandStill;
+    using message::strategy::WalkInsideBoundedBox;
+    using message::strategy::WalkToBall;
+    using message::strategy::WalkToFieldPosition;
+
+    using DefenderTask = message::purpose::Defender;
+    using utility::support::Expression;
+
+    Defender::Defender(std::unique_ptr<NUClear::Environment> environment) : BehaviourReactor(std::move(environment)) {
+
+        on<Configuration>("Defender.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file Defender.yaml
+            this->log_level    = config["log_level"].as<NUClear::LogLevel>();
+            cfg.ready_position = config["ready_position"].as<Expression>();
+        });
+
+        on<Provide<DefenderTask>, Optional<Trigger<GameState>>>().then(
+            [this](const DefenderTask& Defender_task, const std::shared_ptr<const GameState>& game_state) {
+                log<NUClear::DEBUG>("DEFENDER");
+                // Do not use GameController information if force playing or force penalty shootout
+                if (Defender_task.force_playing) {
+                    play();
+                    return;
+                }
+
+                // Check if there is GameState information, and if so act based on the current mode
+                if (game_state) {
+                    switch (game_state->data.mode.value) {
+                        case GameMode::NORMAL:
+                        case GameMode::OVERTIME: emit<Task>(std::make_unique<NormalDefender>()); break;
+                        default: log<NUClear::WARN>("Game mode unknown.");
+                    }
```

Comment:
you don't need this switch case since you're doing the same thing.

Reply:
I believe I do in case of unknown game state

Reply:
if you really want to print something in an unknown game mode then do it in like an if, we don't need a whole switch here.

Reply:
Just check if `game_state->data.mode.value` is NORMAL or OVERTIME and if not then print a warning, otherwise emit the task.

Reply:
I honestly prefer the switch and we will need it for other penalty/gamemode behaviour. 

If statement isn't any cleaner:

```
 // Check if there is GameState information, and if so act based on the current mode
  if (game_state
      && (game_state->data.mode.value == GameMode::NORMAL
          || game_state->data.mode.value == GameMode::OVERTIME)) {
      emit<Task>(std::make_unique<NormalDefender>());
  }
  else {
      log<NUClear::WARN>("Game mode unknown.");
  }
```